src/plugins/intel_gpu/tests/functional/subgraph_tests/reshape_group_convolution.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_test_utils/ov_tensor_utils.hpp"
+#include "common_test_utils/ov_test_utils.hpp"
+#include "openvino/core/coordinate_diff.hpp"
+#include "openvino/core/strides.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/group_conv.hpp"
+#include "openvino/op/parameter.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/opsets/opset13_decl.hpp"
+#include "openvino/pass/manager.hpp"
+#include "openvino/runtime/exec_model_info.hpp"
+#include "shared_test_classes/base/ov_subgraph.hpp"
+#include "intel_gpu/runtime/engine.hpp"
+
+// CreateReshapeOp() for 4d->5d reshape when allow_new_shape_infer=false causes creation of weight reorder with 4d input and 5d output.
```

Comment:
Could you explain why 4d->5d reshape happens in such case? I guess the weights are already in 5d because it is a weight of group conv..